### ISSUE TITLE
option for lane-based routing

### ISSUE THREAD

**Issue by ambuehll** [eclipse-sumo/sumo#15102]
as discussed on matrix:

> quick question on the routing speed for astar or CH in sumo micro vs meso:
> 
> does the router take into account different lane speeds in sumo micro?
> does the router take into account different lane speeds in sumo meso when the approach lane differ and meso-lane is on?
> 

--> According @m-kro the speeds are aggregated over all lanes. I see some issues when a turning lane is completely congested and the through lanes are not. If the average is calculated this also means that the travel time for the left turn movement is underestimated and for the through movement overestimated. 

Would it be possible to have a lane-based routing? 
this is definitely related to #8904

As always, many thanks!

**Comment by namdre** [comment]
you can use option **--weights.separate-turns** to distinguish travel time by turning direction.
That isn't to say a lane-based routing isn't possible but it's definitely a larger piece of work (and no timeline for implementation).

**Comment by ambuehll** [comment]
Ah, right! forgot about that option or actually I thought it was for marouter... 
Thanks for pointing it out. Any suggestion for a value to test with? :)

**Comment by namdre** [comment]
try values in [0.5, 1]

**Comment by ambuehll** [comment]
interestingly, setting the option leads to many more routes taking turn-lanes (which in the end leads to much more congestion). Is that expected? I can provide an example, if interested.

**Comment by namdre** [comment]
No, it's not expected! If you can build a small example that would be great. A single big scenario is too much to analyze for this effect.

**Comment by ambuehll** [comment]
<img width="613" alt="image" src="https://github.com/eclipse-sumo/sumo/assets/48930860/896136b9-65ba-40f6-8e8c-a8f5eb115287">

once weights.separate-turns is set the above route gets very attractive

sumo version: yesterday's dev

[Archive.zip](https://github.com/user-attachments/files/16009540/Archive.zip)
